tvm/src/codegen/opencl/codegen_sdaccel.cc

```diff
@@ -0,0 +1,278 @@
+//       // directly 4 8 bit int in integer.
+//       os << "int"; return;
+//     }
+//     switch (t.bits()) {
+//       case 8: os << "char"; break;
+//       case 16: os << "short"; break;
+//       case 32: os << "int"; break;
+//       case 64: os << "long"; break;
+//       case 1: os << "int"; break;
+//       default: fail = true; break;
+//     }
+//     if (!fail && lanes == 1) return;
+//     if (!fail && (lanes >= 2 && lanes <= 16)) {
+//       os << lanes; return;
+//     }
+//   }
+//   LOG(FATAL) << "Cannot convert type " << t << " to SDAccel type";
+// }
+
+
+void CodeGenSDACCEL::PrintType(Type t, std::ostream& os) {  // NOLINT(*)
+  int lanes = t.lanes();
+  if (t.is_handle()) {
+    //LOG(FATAL) << "The buffer shouldn't call PrintType for printing type";
+    os << "void*";
+    return ;
+  }
+  bool fail = false;
+  if (t.is_float()) {
+    switch (t.bits()) {
+      case 16: os << "half"; break;
+      case 32: os << "float"; break;
+      case 64: os << "double"; break;
+      // case 128: os << "double double"; break;
+      default: fail = true; break;
+    }
+    if (!fail && lanes == 1) return;
+    if (!fail && (lanes >= 2 && lanes <= 16)) {
+      os << lanes; return;
+    }
+  } else if (t.is_uint() || t.is_int()) {
+    if (t.is_uint()) {
+      os << "unsigned ";
+    }
+    if (t.bits() == 8 && t.lanes() == 4) {
+      // directly 4 8 bit int in integer.
+      os << "int"; return;
+    }
+
+    int target_bit = 1;
+    while (target_bit < t.bits())
+      target_bit <<= 1;
+
+    switch (target_bit) {
+      case 1: os << "int"; break;
+      case 2: os << "char"; break;
+      case 4: os << "char"; break;
+      case 8: os << "char"; break;
+      case 16: os << "short"; break;
+      case 32: os << "int"; break;
+      case 64: os << "long"; break;
+      case 128: os << "long"; break; // FIXME: Should use long long
+      default: fail = true; break;
+    }
+    if (!fail && lanes == 1) return;
+    // FIXME: Not yet support multiple lanes
+    //if (!fail && (lanes >= 2 && lanes <= 16)) {
+    //  os << lanes; return;
+    //}
+  }
+  os << t;
+  LOG(WARNING) << "Cannot convert type " << t ;
+  return ;
+}
+
+void CodeGenSDACCEL::PrintStorageScope(
+    const std::string& scope, std::ostream& os) { // NOLINT(*)
+  if (scope == "global" || scope == "shared") {
+    os << "__local ";
+  }
+}
+
+void CodeGenSDACCEL::VisitStmt_(const For* op) {
+  std::ostringstream os;
+  if (op->for_type == ForType::Unrolled) {
+    int unroll_factor = 0, i = 0;
+    for (auto key : op->annotate_keys) {
+      if (auto str = key.as<StringImm>()) {
+        auto factor = op->annotate_values[i].as<IntImm>();
+        if (str->value == "factor" && factor != nullptr && factor->value > 1) {
+          unroll_factor = factor->value;
+          break;
+        }
+      }
+      i++;
+    }
+    if (unroll_factor > 0) {
+        os << "__attribute__((opencl_unroll_hint(";
+        os << unroll_factor << ")))\n";
+    }
+    else

```

Comment:
Please wrap it with {}. Please check [here](https://google.github.io/styleguide/cppguide.html#Conditionals).